CRM/Contribute/BAO/Contribution.php

```diff
@@ -3626,7 +3639,9 @@ public static function recordAdditionalPayment($contributionId, $trxnsData, $pay
       // the reason because it performs 'status change' related code execution for financial records
       // which in 'Pending Refund' => 'Completed' is not useful, instead specific financial record updates
       // are coded below i.e. just updating financial_item status to 'Paid'
-      $contributionDetails = CRM_Core_DAO::setFieldValue('CRM_Contribute_BAO_Contribution', $contributionId, 'contribution_status_id', $statusId);
+      if ($updateStatus) {
```

Comment:
Once you have merged the rows from this line up (preferably with tests)  should not get many more merge conflicts